### PR TITLE
Cls test

### PR DESCRIPTION
**Pull request by carlosggarcia**
Added tests for the `cls` code. Updated `MapperDummy` to work better with it, too, as well as its tests. The Cls are now obtained by inputting some cosmological parameters. The redshift distributions used are in `xcell/tests/data`: `DESY1gc_dndz_bin0.txt` for galaxy clustering and `Nz_DIR_z0.1to0.3.asc` for shear. These come from DESY1 and KV450 datasets.

### CODE REVIEW

File: xcell/tests/test_cls.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    # Compute covariance
+    cov_class = get_cov_class()
+    cov = cov_class.get_covariance()
+
+    # Check that true Cl is within 5sigma of data Cl
+    sigma = np.sqrt(np.diag(cov))
+    cl_p = cl + 5 * sigma
+    cl_m = cl - 5 * sigma
+    check = np.all((cl_m1[0] > cl_m[0]) * (cl_m1[0] < cl_p[0]))
```
Comment (damonge) [inline xcell/tests/test_cls.py:84]: A bit more compact if you do `assert np.all(np.fabs(cl_m1-cl) < 5*sigma)`

File: xcell/tests/test_mapper_dummy.py [hunk 2]
```diff
@@ -30,26 +60,33 @@ def test_get_mask():
...
-    rdev = np.fabs(d / d0 - 1)
+    rdev = np.fabs(d[0] / d0[0] - 1)
+    assert np.max(rdev) < 1E-5
+    # galaxy_shear
+    m = get_mapper('galaxy_shear')
+    d = m.get_signal_map()
+    assert len(d) == 2
+    d0 = [hp.read_map('xcell/tests/data/dummy_signal_s2_0.fits'),
+          hp.read_map('xcell/tests/data/dummy_signal_s2_1.fits')]
+    rdev = np.fabs(d[0][d[0] != 0] / d0[0][d0[0] != 0] - 1)
```
Comment (damonge) [inline xcell/tests/test_mapper_dummy.py:87]: this has a risk of dividing by numbers that may be close to zero, leading to numerical inaccuracies. For this, I usually compare the standard deviation of the difference between both maps with the standard deviation of either of them, and make sure the former is several orders of magnitude smaller than the latter.

**Review by damonge — CHANGES_REQUESTED** [review]

**Comment by damonge** [comment]
@carlosggarcia you also need to install pyccl on github actions. Try adding it [here.](https://github.com/xC-ell/DEScls/blob/97100f8b8618dad19a57d2fcc8778e462d20c828/.github/workflows/ci.yml#L41)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -39,6 +39,7 @@ jobs:
             pip install -U healpy
             pip install -U pandas
             pip install -U pymaster
+            pip install -U pyccl
```
Comment (damonge) [inline .github/workflows/ci.yml:42]: Try adding `pip install -U camb`

File: xcell/mappers/mapper_dummy.py [hunk 4]
```diff
@@ -1,50 +1,125 @@
...
-           'mask_name': ***}
+           'cosmo': {
+                'Omega_c': 0.2640,
+                'Omega_b': 0.0493,
+                'h': 0.6736,
+                'n_s': 0.9649,
+                'sigma8': 0.8111,
+                'w0': -1,
+                'wa': 0,
+                'transfer_function': 'boltzmann_class',
```
Comment (damonge) [inline xcell/mappers/mapper_dummy.py:22]: And change this to `boltzmann_camb`

File: xcell/mappers/mapper_dummy.py [hunk 5]
```diff
@@ -1,50 +1,125 @@
...
+            # Planck 2018: Table 2 of 1807.06209
+            # Omega_m: 0.3133
+            'Omega_c': 0.2640,
+            'Omega_b': 0.0493,
+            'h': 0.6736,
+            'n_s': 0.9649,
+            'sigma8': 0.8111,
+            'w0': -1,
+            'wa': 0,
+            'transfer_function': 'boltzmann_class',
```
Comment (damonge) [inline xcell/mappers/mapper_dummy.py:41]: and this

File: xcell/tests/test_cls.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        # Planck 2018: Table 2 of 1807.06209
+        # Omega_m: 0.3133
+        'Omega_c': 0.2640,
+        'Omega_b': 0.0493,
+        'h': 0.6736,
+        'n_s': 0.9649,
+        'sigma8': 0.8111,
+        'w0': -1,
+        'wa': 0,
+        'transfer_function': 'boltzmann_class',
```
Comment (damonge) [inline xcell/tests/test_cls.py:27]: and this

File: xcell/tests/test_mapper_dummy.py [hunk 7]
```diff
@@ -1,17 +1,47 @@
...
+def get_config(dtype='galaxy_density'):
+    cosmo = {
+      'Omega_c': 0.24,
+      'Omega_b': 0.0493,
+      'h': 0.72,
+      'n_s': 0.9649,
+      'sigma8': 0.78,
+      'w0': -1,
+      'wa': 0,
+      'transfer_function': 'boltzmann_class',
```
Comment (damonge) [inline xcell/tests/test_mapper_dummy.py:16]: and this

**Review by damonge — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (carlosggarcia) [reply .github/workflows/ci.yml:42]: Done

File: xcell/tests/test_mapper_dummy.py (hunk 2 above)
Comment (carlosggarcia) [reply xcell/tests/test_mapper_dummy.py]: Done

**Review by damonge — APPROVED** [review]

**Pull request merged**